app/genesis_integration_test.go

```diff
@@ -0,0 +1,86 @@
+package app
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	poetypes "github.com/confio/tgrade/x/poe/types"
+
+	"github.com/stretchr/testify/assert"
+	tmjson "github.com/tendermint/tendermint/libs/json"
+
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	db "github.com/tendermint/tm-db"
+
+	"github.com/confio/tgrade/x/twasm"
+	twasmtypes "github.com/confio/tgrade/x/twasm/types"
+)
+
+func TestTgradeGenesisExportImport(t *testing.T) {
+	doInitWithGenesis := func(gapp *TgradeApp, genesisState GenesisState) {
+		setupWithSingleValidatorGenTX(t, genesisState)
+
+		stateBytes, err := json.MarshalIndent(genesisState, "", "  ")
+		require.NoError(t, err)
+
+		// Initialize the chain
+		gapp.InitChain(
+			abci.RequestInitChain{
+				Time:          time.Now().UTC(),
+				Validators:    []abci.ValidatorUpdate{},
+				AppStateBytes: stateBytes,
+			},
+		)
+		gapp.Commit()
+	}
+	memDB := db.NewMemDB()
+	gapp := NewTgradeApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), memDB, nil, true, map[int64]bool{}, DefaultNodeHome, 0, MakeEncodingConfig(), EmptyBaseAppOptions{}, emptyWasmOpts)
+	initGenesis := NewDefaultGenesisState()
+
+	doInitWithGenesis(gapp, initGenesis)
+	// Making a new app object with the db, so that initchain hasn't been called
+	newGapp := NewTgradeApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), memDB, nil, true, map[int64]bool{}, DefaultNodeHome, 0, MakeEncodingConfig(), EmptyBaseAppOptions{}, emptyWasmOpts)
```

Comment:
Wait, sharing the same memDB? That looks odd..

Why can't you export with gapp?

(Note, I would make a block or two on app before the export to be a bit more realistic. Even if they are empty or simple bank SendMsg)